backend/baseclass.pm

```diff
@@ -889,20 +890,25 @@ sub wait_serial {
     }
 
     $regexp = [$regexp] if ref $regexp ne 'ARRAY';
-    my $initial_time = time;
-    while (time < $initial_time + $timeout) {
+    my $initial_time   = time;
+    my $current_offset = $self->{serial_offset};
+  MATCH: while (time < $initial_time + $timeout) {
```

Comment:
I would prefer to not use goto-like statements. Can we find a better approach?

Reply:
I can check `$matched && last` could work, too. Lemme try